### ISSUE TITLE
EPIC: Scale OSIO analytics platform 

### ISSUE THREAD

**Issue by krishnapaparaju** [openshiftio/openshift.io#1144]
Need to scale OSIO analytics to process 100 stack analysis requests (SLA: 2 mins) , 500 components analysis (SLA: 1 sec) requests concurrently. 

This work is being tracked here and all the code/configuration changes that would be made to achieve the above goal around scalability.

https://docs.google.com/a/redhat.com/spreadsheets/d/13CRoYJPxGrDkhgnTKeyjP80SVdY0Sdfg6bbGlIT6azY/

cc @msrb @samuzzal-choudhury @msrb @GeorgeActon @sivaavkd

**Comment by fridex** [comment]
> 500 components analysis (SLA: 1 sec) requests concurrently.

How is this defined? 1 sec for one component analysis?

**Comment by krishnapaparaju** [comment]
@fridex Yes, OSIO user triggering component analysis would expect the response in 1 second and response for stack analysis in not more than 2 mins.

**Comment by fridex** [comment]
> @fridex Yes, OSIO user triggering component analysis would expect the response in 1 second and response for stack analysis in not more than 2 mins.

Is it response from API server in both cases?

**Comment by krishnapaparaju** [comment]
Yes

**Comment by fridex** [comment]
> Yes

OK, thanks for clarification.

**Comment by krishnapaparaju** [comment]
@samuzzal-choudhury @sivaavkd @GeorgeActon scaling the platform for significantly higher number of users is on the critical path for opening up the platform for IDE users.

**Comment by msrb** [comment]
Component analyses are OK-ish (not flawless though): https://github.com/openshiftio/openshift.io/issues/1243#issuecomment-342399807

There will be some work around stack analyses: https://github.com/openshiftio/openshift.io/issues/1242#issuecomment-342418630
